dango/types/src/oracle/pyth.rs

```diff
@@ -0,0 +1,265 @@
+    vaa: WormholeVaa,
+    feeds: Vec<PriceFeed>,
+}
+
+impl PythVaa {
+    pub fn verify(
+        self,
+        storage: &dyn Storage,
+        api: &dyn Api,
+        block: BlockInfo,
+        guardian_set: Map<u32, GuardianSetInfo>,
+    ) -> anyhow::Result<Vec<PriceFeed>> {
+        self.vaa.verify(storage, api, block, guardian_set)?;
+        Ok(self.feeds)
+    }
+
+    pub fn unverified(self) -> Vec<PriceFeed> {
+        self.feeds
+    }
+
+    pub fn new<T>(bytes: T) -> anyhow::Result<Self>
+    where
+        T: Into<Vec<u8>>,
+    {
+        let bytes = bytes.into();
+
+        let (vaa, feeds) = if bytes[0..4] == *PYTHNET_ACCUMULATOR_UPDATE_MAGIC {
+            let res = AccumulatorUpdateData::try_from_slice(&bytes)?;
+
+            let (vaa, updates) = uncast_enum!(res.proof, Proof::WormholeMerkle, vaa, updates);
+
+            let parsed_vaa = WormholeVaa::new(Vec::from(vaa))?;
+            let msg = WormholeMessage::try_from_bytes(parsed_vaa.payload.clone())?;
+
+            let root = MerkleRoot::<Keccak160>::new(
+                uncast_enum!(msg.payload, WormholePayload::Merkle).root,
+            );
+
+            let feeds = updates
+                .into_iter()
+                .map(|update| {
+                    let message_vec = Vec::from(update.message);
+
+                    if !root.check(update.proof, &message_vec) {
+                        bail!("invalid proof");
+                    }
+
+                    let msg = from_slice::<BigEndian, Message>(&message_vec)?;
+
+                    let price_feed = match msg {
+                        Message::PriceFeedMessage(price_feed_message) => PriceFeed::new(
+                            PriceIdentifier::new(price_feed_message.feed_id),
+                            Price {
+                                price: price_feed_message.price,
+                                conf: price_feed_message.conf,
+                                expo: price_feed_message.exponent,
+                                publish_time: price_feed_message.publish_time,
+                            },
+                            Price {
+                                price: price_feed_message.ema_price,
+                                conf: price_feed_message.ema_conf,
+                                expo: price_feed_message.exponent,
+                                publish_time: price_feed_message.publish_time,
+                            },
+                        ),
+                        _ => bail!("invalid message"),
+                    };
+
+                    Ok(price_feed)
+                })
+                .collect::<anyhow::Result<Vec<_>>>()?;
+
+            (parsed_vaa, feeds)
+        } else {
+            let vaa = WormholeVaa::new(bytes)?;
+            let batch_attestation =
+                BatchPriceAttestation::deserialize(&vaa.payload[..]).map_err(|err| {
+                    anyhow::anyhow!("Failed to deserialize batch attestation: {:?}", err)
```

Comment:
We need to be more consistent on how to write error messages. Earlier in this file you write "invalid message" which is all lowercase. But here you use "Failed to..." which starts with an uppercase.

Let's use all lowercase everywhere, which is what rust style guide suggests: https://github.com/rust-lang/api-guidelines/issues/79